tests/test_container.py

```diff
@@ -137,3 +138,74 @@ def factory():
         assert close_me.is_closed
         assert not container._instantiated
         assert not container._on_close
+
+    def test_local_pings_are_retrieved(self, registry, container):
+        """
+        Registering a local factory with a ping defined should make it possible to
+        invoke a ping for that service.
+        """
+
+        def another_service_factory():
+            yield AnotherService()
+
+        another_service_ping = Mock(spec_set=["__call__"])
+        container.register_local_factory(
+            AnotherService, another_service_factory, ping=another_service_ping
+        )
+
+        (svc_ping,) = container.get_pings()
+        svc_ping.ping()
+
+        another_service_ping.assert_called_once()
+
+    def test_local_pings_override_global_pings(self, registry, container):
+        """
+        If a local factory overwrites an existing, global one, and the local factory has a ping
+        defined, the local ping should be used.
+        """
+
+        def another_service_factory():
+            yield AnotherService()
+
+        another_service_ping = Mock(spec_set=["__call__"])
+        local_another_service_ping = Mock(spec_set=["__call__"])
+        registry.register_factory(
+            AnotherService, another_service_factory, ping=another_service_ping
+        )
+        container.register_local_factory(
+            AnotherService,
+            another_service_factory,
+            ping=local_another_service_ping,
+        )
+
+        (svc_ping,) = container.get_pings()
+        svc_ping.ping()
+
+        another_service_ping.assert_not_called()
+        local_another_service_ping.assert_called_once()
+
+    def test_local_services_without_pings_do_not_discard_global_pings(
```

Comment:
I assumed such behavior would be the most "predictable" from the end-user perspective, but I was not quite sure about this change. Would you find it valid to make it work like that?

Reply:
yeah hm this is hard. There's two contradicting issues here:

1. discarding is a breaking change
2. If I overwrite a definition, _that's_ what I want to use.

But more specifically, it's the decision of the _user_ to add a new definition of a service with a ping. I feel like discarding follows the wish of the user? They can define a new type based on the old type if they want to preserve the old definition.

Given that *svcs* core concept are types, I feel like we should respect it when a user says "Connection is now _this_ new thing."

But don't change anything, yet.

Reply:
Ok, so regardless of whether a new definition contains a ping, the latest defined service (could be local) must be used?